### ISSUE TITLE
About type error(triggers by datalodaer)

### ISSUE THREAD

**Issue by JohnW2all** [wangzt-halo/das#6]
Environment：
![image](https://user-images.githubusercontent.com/75520758/218623878-201b6f45-b80e-473e-9efb-34d26217ab1c.png)

Dataset：
![image](https://user-images.githubusercontent.com/75520758/218623992-6bf4ada7-1e1e-4b91-b572-05a52959c466.png)
![image](https://user-images.githubusercontent.com/75520758/218624040-da97efd2-6bc9-48ae-b51b-939292a562e2.png)
![image](https://user-images.githubusercontent.com/75520758/218624254-24a9f629-7981-4909-9580-dd7ca045d93c.png)
![image](https://user-images.githubusercontent.com/75520758/218624297-a83d8ec7-9b43-4aa4-ac81-c465dbe532de.png)
![image](https://user-images.githubusercontent.com/75520758/218624328-2fb9ed99-4d3b-4c19-8471-cd2a33f2665f.png)

Using a testing command “python tools/test.py configs/das/exp_mupots.py weights/latest.pth --eval pck”, triggering an error:
![image](https://user-images.githubusercontent.com/75520758/218624834-04093bc0-35cc-46b9-b9d8-9d3048f76436.png)

**Comment by wangzt-halo** [comment]
`pose_vis` should be an array contained in the annotation file. Could you print out its value?

**Comment by JohnW2all** [comment]
> `pose_vis` should be an array contained in the annotation file. Could you print out its value?
![image](https://user-images.githubusercontent.com/75520758/219369248-d2a49dd8-bbfc-4a02-a696-910be5f84aaa.png)

The value of **pose_vis** is "True", so the last sentence in the last picture says"TypeError: ' bool' object is not subscriptable".
I have not modified the source code, so I wonder if there is any problem with the code?
Thanks!

**Comment by JohnW2all** [comment]
> `pose_vis` should be an array contained in the annotation file. Could you print out its value?

The value of pose_vis is "True", so the last sentence in the last picture says"TypeError: ' bool' object is not subscriptable".
![image](https://user-images.githubusercontent.com/75520758/219370707-d4f7b041-8368-41aa-90e6-74bc9c176b90.png)

Then I print the "ann['keypoints_vis']", it's an array full of "True".
![image](https://user-images.githubusercontent.com/75520758/219372457-74593ac0-385d-4a4e-8e2c-a43f918e51c1.png)
![image](https://user-images.githubusercontent.com/75520758/219372372-428739ae-b2e0-4cfc-82a0-3266dcc2c87d.png)

Then I print the "ann_info".
![image](https://user-images.githubusercontent.com/75520758/219536895-857892a3-bc63-4b7f-a57b-0bb9bb51f7f9.png)
![image](https://user-images.githubusercontent.com/75520758/219536794-24be6994-c322-493c-8e01-0ebfdb2f620b.png)


I haven't changed the source code, so I'm confused about what went wrong.
I would like to ask if there are any errors in my coco dataset folder?
![image](https://user-images.githubusercontent.com/75520758/219374061-7e5976fc-1d72-4666-bbcf-5cc10b60b213.png)
![image](https://user-images.githubusercontent.com/75520758/219374104-6269395e-6f18-4477-a74f-96e6e7bfb122.png)


Thanks!

**Comment by wangzt-halo** [comment]
ok. you can try to replace the code `if pose_vis[self.ROOT_IDX][0] == 0:` with `if pose_vis[self.ROOT_IDX] == 0:` and see if this would work.

**Comment by JohnW2all** [comment]
> ok. you can try to replace the code `if pose_vis[self.ROOT_IDX][0] == 0:` with `if pose_vis[self.ROOT_IDX] == 0:` and see if this would work.

I truly appreciate your timely help. It works!

After this a few small errors appear, now I am trying to fix it.
![image](https://user-images.githubusercontent.com/75520758/219661754-5e411411-dcfe-4dca-bbbc-18aa625f62b9.png)

**Comment by JohnW2all** [comment]
> ok. you can try to replace the code `if pose_vis[self.ROOT_IDX][0] == 0:` with `if pose_vis[self.ROOT_IDX] == 0:` and see if this would work.

Excuse me，I met a error about SVD in the following picture.
![image](https://user-images.githubusercontent.com/75520758/219848672-730794c6-32a7-48a3-a601-10c97a4b6e12.png)

I printed the H and all the variables associated with it.
![image](https://user-images.githubusercontent.com/75520758/219848824-9110e543-9811-49bd-b99e-65734041b446.png)
![image](https://user-images.githubusercontent.com/75520758/219848779-31158664-0f43-4211-b783-e08a4d19aed3.png)
There are **Nan** values among them. May I ask if this is the cause of the problem?

**Comment by JohnW2all** [comment]
> ok. you can try to replace the code `if pose_vis[self.ROOT_IDX][0] == 0:` with `if pose_vis[self.ROOT_IDX] == 0:` and see if this would work.

Excuse me，I met a error about SVD and I don't know how to deal with this problem.
![image](https://user-images.githubusercontent.com/75520758/219936091-9308dbeb-e5b2-4ae2-abe1-e7f9e1485cff.png)

I printed the H and all the variables associated with it.
![image](https://user-images.githubusercontent.com/75520758/219936096-f652f868-af59-49e4-9df3-a699221ccfdd.png)
![image](https://user-images.githubusercontent.com/75520758/219936107-326f99cf-ef10-4890-8c47-bf3b9123b615.png)

There are **nan** values among them. May I ask if this is the cause of the problem and could you please tell me how to fix it?
Thanks so much!

**Comment by wangzt-halo** [comment]
It seems that the problem comes from the wrong predictions. Are the outputs all `nan'  for every data sample? If so, maybe there is something wrong with the trained model.

**Comment by JohnW2all** [comment]
Yes, all the output variable **predict** in the terminal contains **nan**.
I noticed that the terminal told me that the model and the weights did not match, which may have caused the problem.
![image](https://user-images.githubusercontent.com/75520758/219989346-058ece99-c0d5-445e-b536-276a89343d2f.png)

Oh, I think I see the cause of the problem. Because I use the "latest.pth" provided by you, which is trained in the CMU Panoptic dataset.
![image](https://user-images.githubusercontent.com/75520758/219989892-074fd462-d651-4ecd-be14-082927b78b45.png)
And I use it incorrectly to predict in MuPoTS-3D dataset.

Thanks!

**Comment by tribeband** [comment]
I am not able to download the muco/human3.6m data from the link provided. will you please help me to provide another link?

**Comment by wangzt-halo** [comment]
@tribeband Because this data does not belong to us, we are sorry that we cannot provide you with another link. 
But I checked the original link and it is still available for download.

**Comment by tribeband** [comment]
曾经我下载过，但是下完之后解压出错
